### PR TITLE
Don't auto-declare override parameters and fix some prints

### PR DESCRIPTION
**Pull request by bmagyar**
https://github.com/ros-controls/ros2_controllers/issues/129

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -318,7 +318,8 @@ controller_interface::return_type ControllerManager::configure_controller(
     RCLCPP_ERROR(
       get_logger(),
       "after configuring, controller %s is in state %s, expected Inactive",
```
Comment (AndyZe) [inline controller_manager/src/controller_manager.cpp:320]: nitpick - capitalize After

**Review by AndyZe — APPROVED** [review]

File: controller_manager/src/controller_manager.cpp (hunk 1 above)
Comment (destogl) [reply controller_manager/src/controller_manager.cpp:320]: Good catch!

There are many of those "capitalization" issues. We should at least decide if we start with small or capitalized letters.

File: controller_interface/src/controller_interface.cpp [hunk 2]
```diff
@@ -28,8 +28,7 @@ ControllerInterface::init(const std::string & controller_name)
 {
   node_ = std::make_shared<rclcpp::Node>(
     controller_name,
-    rclcpp::NodeOptions().allow_undeclared_parameters(true).
-    automatically_declare_parameters_from_overrides(true));
+    rclcpp::NodeOptions().allow_undeclared_parameters(true));
```
Comment (destogl) [inline controller_interface/src/controller_interface.cpp:31]: If you remove the second line there were issues with not auto-declaring / auto-setting of parameters. Please check against demo repository before merging.

File: controller_manager/src/controller_manager.cpp [hunk 3]
```diff
@@ -318,7 +318,8 @@ controller_interface::return_type ControllerManager::configure_controller(
...
-      controller_name.c_str());
+      controller_name.c_str(),
+      state.label().c_str());
```
Comment (destogl) [inline controller_manager/src/controller_manager.cpp:322]: `new_state` is expected here...

```suggestion
      new_state.label().c_str());
```

File: hardware_interface/src/resource_manager.cpp [hunk 4]
```diff
@@ -227,13 +227,13 @@ LoanedCommandInterface ResourceManager::claim_command_interface(const std::strin
 {
   if (!command_interface_exists(key)) {
     throw std::runtime_error(
-            std::string("command interface with key") + key + " does not exist");
+            std::string("Command interface with ") + key + " does not exist");
```
Comment (destogl) [inline hardware_interface/src/resource_manager.cpp:230]: is so, than as @AndyZe commented, capitalize all output.

**Review by destogl — COMMENTED** [review]
Please confirm that the first change is really working. I had some issues with that earlier...

File: controller_interface/src/controller_interface.cpp (hunk 2 above)
Comment (bmagyar) [reply controller_interface/src/controller_interface.cpp:31]: That's how I actually found this issue. Anything that uses a more complex parameter setup can't work with the original set of flags.

I think this used to be here as a lucky charm when Dashing had issues with param handling. It is supposed to allow auto-declaring parameters for a node when they are set from outside but the `automatically_declare_parameters_from_overrides` flag forces those to be declared, when we re-declare it inside `init()`, there's a clash. If we want to pre-load yaml configurations for controllers that also have defaults if unspecified, we need to remove that line.

Comment (bmagyar) [reply controller_interface/src/controller_interface.cpp:31]: https://github.com/ros-controls/ros2_controllers/pull/131

**Pull request merged**